proc/threads.go

```diff
@@ -280,6 +280,12 @@ func (thread *Thread) GetG() (g *G, err error) {
 		return nil, err
 	}
 
+	if thread.dbp.arch.GStructOffset() == 0 {
+		// GetG was called through SwitchThread / updateThreadList during initialization
+		// thread.dbp.arch isn't setup yet (it needs a CurrentThread to read global variables from)
+		return nil, nil
```

Comment:
Was this put in so that `GetG` calls in `SwitchThread` would work fine, and essentially set `SelectedGoroutine` to nil if we don't have a proper `GStructOffset`?